ext/crypto/lib.rs

```diff
@@ -789,8 +790,32 @@ pub async fn op_crypto_derive_bits(
 pub struct EncryptArg {
   key: KeyData,
   algorithm: Algorithm,
+  // RSA-OAEP
   hash: Option<CryptoHash>,
   label: Option<ZeroCopyBuf>,
+  // AES-CBC
+  iv: Option<ZeroCopyBuf>,
+  length: Option<usize>,
+}
+
+macro_rules! cipher_cbc {
+  ($length: ty, $key: expr, $iv: expr, $data: expr) => {{
+    // Section 10.3 Step 2 of RFC 2315 https://www.rfc-editor.org/rfc/rfc2315
+    type AesCbc = block_modes::Cbc<$length, block_modes::block_padding::Pkcs7>;
+    let cipher = AesCbc::new_from_slices($key, $iv)?;
```

Comment:
Macros with control flow in them (the ? operator) are a special kind of hell to debug. Please either use combinators (and_then or map) or turn them into regular functions.

Reply:
Made them inline it's not much duplicate code anyways.